### ISSUE TITLE
PublicMethodsHaveTypeChecker should not flag nested defs

### ISSUE THREAD

**Issue by megri** [scalastyle/scalastyle#48]
The following snippet has **def bar** flagged as a public method with no explicit return type

``` scala
  def foo(n: Int): Int = {
    def bar = n + 1
    bar
  }
```

**Comment by matthewfarwell** [comment]
Yes, this should be raised as a separate issue. It is a relatively easy
fix, which I'll do when I get a minute.

Have fun,

Matthew Farwell.

2014-12-29 21:01 GMT+01:00 Leif Wickland <notifications@github.com>:

**Comment by matthewfarwell** [comment]
#129 has now been fixed, Could someone test it and see if it really does fix their issues please?

Thanks.

**Comment by leifwickland** [comment]
@matthewfarwell, Yes #129 is fixed for me. Thanks for your work.
